lib/segment/src/vector_storage/simple_sparse_vector_storage.rs

```diff
@@ -172,7 +174,12 @@ impl VectorStorage for SimpleSparseVectorStorage {
     }
 
     fn get_vector(&self, key: PointOffsetType) -> CowVector {
-        self.get_sparse(key).into()
+        self.get_vector_opt(key).expect("Vector must exist")
+    }
```

Comment:
Should we return an error here, rather than panicing? Or can't we easily propagate such error in the places we use this.

Reply:
There is a possible broader refactoring  to propagate missing key or data corruption error properly.
But this PR must be very targeted for the release, I 'd rather focus on the issue at hand.